### PR TITLE
feat(core): enable root project by specifying `nx: true` in root package.json

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
Root package scripts are included if an env variable is provided.

## Expected Behavior
Root package scripts are automatically included if the `nx` field is present in the root package.json. You can set it to `true` if none of the configurations inside it is needed. 

The nx property can also be set to `false` to ignore any package.json's which should not be considered.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/generators/utils/project-configuration.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { PackageJson } from 'nx/src/utils/package-json';
```
Comment (vsavkin) [inline packages/nx/src/generators/utils/project-configuration.ts:1]: Should use rleative imports

File: packages/nx/src/generators/utils/project-configuration.ts [hunk 2]
```diff
@@ -435,24 +436,29 @@ function inlineProjectConfigurationsWithTree(
...
+      const fileName = basename(change.path);
+      // all created project json files are created projects
+      if (fileName === 'project.json') {
+        createdProjectFiles.push(change.path);
+      } else if (fileName === 'package.json') {
+        // created package.json files are projects by default *unless* they are at the root
+        const includedByDefault = change.path === 'package.json' ? false : true;
+        const contents: PackageJson = JSON.parse(change.content.toString());
+        // If the file should be included by default, and is not excluded, or it is included by the `nx` property
+        if ((includedByDefault && contents.nx !== false) || contents.nx) {
```
Comment (vsavkin) [inline packages/nx/src/generators/utils/project-configuration.ts:455]: why did you add nx !== false? This didn't exist before and shouldn't exist

File: packages/nx/src/generators/utils/project-configuration.ts [hunk 3]
```diff
@@ -435,24 +436,29 @@ function inlineProjectConfigurationsWithTree(
...
-  return deduplicateProjectFiles(files);
+  const createdProjectFiles = [];
+
+  for (const change of tree.listChanges()) {
+    if (change.type === 'CREATE') {
+      const fileName = basename(change.path);
+      // all created project json files are created projects
+      if (fileName === 'project.json') {
+        createdProjectFiles.push(change.path);
+      } else if (fileName === 'package.json') {
```
Comment (vsavkin) [inline packages/nx/src/generators/utils/project-configuration.ts:450]: if you remove support for nx: false, the whole thing can be simplified

File: packages/nx/src/migrations/update-15-0-0/migrate-to-inputs.ts [hunk 4]
```diff
@@ -90,11 +90,12 @@ export default async function (tree: Tree) {
               tree,
               join(project.root, 'package.json'),
               (json) => {
-                json.nx ??= {};
-                json.nx.namedInputs ??= {};
-                json.nx.namedInputs.projectSpecificFiles ??=
+                const packageJsonNxConfiguration =
+                  typeof json.nx === 'boolean' ? {} : json.nx ?? {};
```
Comment (vsavkin) [inline packages/nx/src/migrations/update-15-0-0/migrate-to-inputs.ts:94]: I don't think the whole nx being a boolean thing

**Review by vsavkin — DISMISSED** [review]
1. nx property should not be a boolean
2. there are no unit tests  or e2e tests

File: packages/nx/src/migrations/update-15-0-0/migrate-to-inputs.ts (hunk 4 above)
Comment (AgentEnder) [reply packages/nx/src/migrations/update-15-0-0/migrate-to-inputs.ts:94]: Let's chat about this one Monday morning

**Review by FrozenPandaz — APPROVED** [review]
